framework/src/Volo.Abp.Guids/Volo/Abp/Guids/SequentialGuidGenerator.cs

```diff
@@ -92,8 +100,119 @@ public Guid Create(SequentialGuidType guidType)
 
                 // For sequential-at-the-end versions, we copy the random data first,
                 // followed by the timestamp.
-                Buffer.BlockCopy(randomBytes, 0, guidBytes, 0, 10);
-                Buffer.BlockCopy(timestampBytes, 2, guidBytes, 10, 6);
+                Buffer.BlockCopy(randomBytes, 0, guidBytes, 0, 8);
+
+                // MSSQL and System.Data.SqlTypes.SqlGuid sort first by the last 6 Data4 bytes, left to right,
+                // then the first two bytes of Data4 (again, left to right),
+                // then Data3, Data2, and Data1 right to left.
+                Buffer.BlockCopy(timestampBytes, 6, guidBytes, 8, 2);
+                Buffer.BlockCopy(timestampBytes, 0, guidBytes, 10, 6);
+                break;
+        }
+
+        return new Guid(guidBytes);
+    }
+
+    /// <summary>
+    /// Generate sequential Guids that conform to the RFC 4122 Version 4.
+    /// </summary>
+    /// <returns></returns>
+    public virtual Guid Next()
+    {
+        return Next(Options.GetDefaultSequentialGuidType());
+    }
+
+    /// <summary>
+    /// Generate sequential Guids that conform to the RFC 4122 Version 4.
+    /// </summary>
+    /// <param name="guidType"></param>
+    /// <returns></returns>
+    public virtual Guid Next(SequentialGuidType guidType)
+    {
+        // According to RFC 4122 Version 4:
+        // dddddddd-dddd-Mddd-Ndrr-rrrrrrrrrrrr
+        // - M = RFC version, in this case '4' for random UUID
+        // - N = RFC variant (plus other bits), where N is one of 8,9,A, or B, in this case 8 for variant 1
+        // - d = nibbles based on UTC date/time in ticks
+        // - r = nibbles based on random bytes
+
+        byte version = (byte)4;
+        byte variant = (byte)8;
```

Comment:
The `Next` method only exists in the implementation class, If you want to expose it we need to update the  `IGuidGenerator` interface.

Reply:
I expose the new method `Next` in the hope that it will generate a new GUID compliant with RFC 4122.